src/lib/mqtt.ts

```diff
@@ -0,0 +1,39 @@
+
+export type MqttSubscriptionCallback = (payload: any, topic: string) => void
+
+export class MqttSubscription {
+    topic: string;
+    regex: RegExp;
+    jsonPayload: boolean;
+    callback: MqttSubscriptionCallback;
+
+    constructor(topic: string, jsonPayload: boolean, callback: MqttSubscriptionCallback) {
+        this.topic = topic;
+        let tmp: string = topic.split("+").join("[^\/]*");
```

Comment:
Please add a comment and maybe a sample of what this regex/string manipulation does.
I don't want to think about it.